### PR TITLE
Various fixes [Description in each commit].

### PR DESCRIPTION
**Pull request by saurabhshri**

### CODE REVIEW

File: mod_auth/forms.py [hunk 1]
```diff
@@ -2,45 +2,48 @@
...
 from wtforms.validators import DataRequired, Email, ValidationError
+from email_validator import validate_email, EmailNotValidError
+
 
 def valid_password(form, field):
     pass_size = len(field.data)
     if pass_size == 0:
         raise ValidationError('The password can not be empty!')
-    if pass_size < 8 or pass_size > 50:
+    if pass_size < 8 or pass_size > 128:
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: I'd extract both the min & max length into a variable and then also put those in the error message. Easier to maintain.

File: tests/test_authentication.py [hunk 2]
```diff
@@ -29,12 +34,29 @@ def test_blank_email(self):
...
+                                 '#@%^%#$@#$@#.com',
+                                 '@domain.com',
+                                 'Joe Smith <email@domain.com>',
+                                 'email.domain.com',
+                                 'mail@domain@domain.com',
+                                 '.email@domain.com',
+                                 'email.@domain.com',
+                                 'email..email@domain.com',
+                                 'email@domain.com (Joe Smith)',
+                                 'email@domain',
```
Comment (canihavesomecoffee) [inline tests/test_authentication.py]: Technically that's a valid email ;)

https://isemail.info/email%40domain

File: tests/test_authentication.py [hunk 3]
```diff
@@ -29,12 +34,29 @@ def test_blank_email(self):
...
+                                 'email@-domain.com',
+                                 'email@domain.web',
```
Comment (canihavesomecoffee) [inline tests/test_authentication.py]: So is this (strictly speaking):

https://isemail.info/email%40domain.web

File: tests/test_authentication.py [hunk 4]
```diff
@@ -29,12 +34,29 @@ def test_blank_email(self):
...
+                                 'email@111.222.333.44444',
```
Comment (canihavesomecoffee) [inline tests/test_authentication.py]: And this ;)

https://isemail.info/email%40111.222.333.44444

**Comment by saurabhshri** [comment]
I used [this](https://blogs.msdn.microsoft.com/testing123/2009/02/06/email-address-test-cases/) to use as the test cases. It has explanation why each email in that list is invalid.

Regarding `email@domain`, I understand that it could be valid for email addresses such as `user@localhost`, but the question is - will we allow users to sign up through such addresses?

>and then just require a user to click a registration link in the email

I agree on this approach. I just need to set up the mailing system or use an API such as mailgun, which I'll try reading and setting. Once it is set, I will add email verification and password reset.

File: mod_auth/forms.py (hunk 1 above)
Comment (saurabhshri) [reply mod_auth/forms.py]: Added.

**Comment by canihavesomecoffee** [comment]
If the system is set up in a good way - that is, send an email with link to complete registration - you don't need to care how the email address looks. If they can receive it, it's good enough :)

Again, for example foo@com might not work right now, but if the guys that operate the com tld decide to put a mailserver on the tld, it's a perfectly valid email address. Hence, if they can receive an email and click the link in there, the email address should be considered valid ;)

**Comment by saurabhshri** [comment]
Gotcha! Once the email verification is set-up, I'll remove these extra email checks.

**Pull request merged**